generators/add-web-assets/exc-react/templates/src/utils.js

```diff
@@ -15,6 +15,19 @@ import actions from './config.json'
 
 /* global fetch */
 
+/**
+ *
+ * Attempts to load the Adobe Experience Cloud Runtime.
+ *
+ * @throws {Error} error in case of failure, most likely when the app is not running in
+ * the Experience Cloud Shell.
+ *
+ */
+function loadExcRuntime () {
+  /* eslint-disable-next-line */
+  (function(e,t){if(t.location===t.parent.location)throw new Error("Module Runtime: Needs to be within an iframe!");var o=function(e){var t=new URL(e.location.href).searchParams.get("_mr");return t||!e.EXC_US_HMR?t:e.sessionStorage.getItem("unifiedShellMRScript")}(t);if(!o)throw new Error("Module Runtime: Missing script!");if("https:"!==(o=new URL(decodeURIComponent(o))).protocol)throw new Error("Module Runtime: Must be HTTPS!");if(!/experience(-qa|-stage)?\.adobe\.com$/.test(o.hostname)&&!/localhost\.corp\.adobe\.com$/.test(o.hostname))throw new Error("Module Runtime: Invalid domain!");if(!/\.js$/.test(o.pathname))throw new Error("Module Runtime: Must be a JavaScript file!");t.EXC_US_HMR&&t.sessionStorage.setItem("unifiedShellMRScript",o.toString());var n=e.createElement("script");n.async=1,n.src=o.toString(),n.onload=n.onreadystatechange=function(){n.readyState&&!/loaded|complete/.test(n.readyState)||(n.onload=n.onreadystatechange=null,n=void 0,"EXC_MR_READY"in t&&t.EXC_MR_READY())},e.head.appendChild(n)})(document, window)
```

Comment:
wouldn't this script be better loaded off a file, for ease of update? unless there are bundler issues

Reply:
no bundler issues, just wanted to limit the number of files the user sees.
but that's a fair point mentioned by @purplecabbage  too, so I guess that's a 2v1 😄 , will do put it back to the separate file